### DISCUSSION TITLE
Pass extraneous metadata as an input?

### DISCUSSION THREAD

**Discussion by gregbenz** [google/libultrahdr#141] [Ideas]
When building with sources which do not support metadata (raw inputs), it would be ideal to have a way to embed metadata in the parent image (such as copyright, title, capture data like shutter speed, etc). This can be achieved with other libraries of course, but if there is an efficient way to merge such an input with the required gain map metadata from libultrahdr, that would be a useful feature and probably ideal for performance at scale.

If this is out of scope, I understand, but want to raise it as something which would be valuable.

**Comment by ram-mohan** [comment]
@gregbenz this can be achieved via uhdr_enc_set_exif_data(). This is a write through api where the library accepts the byte buffer received and write into jpeg as-is. The library does not validate if the byte buffer contents are in compliance with exif file format. Individual configuration of various exif tags is not supported as this is cumbersome and outside the scope of the library

**Reply by gregbenz** [reply]
@ram-mohan Thank you. Would it be possible to expose that in ultrahdr_app?

**Reply by ram-mohan** [reply]
We will try to add an option to accept exif info in bin format

**Comment by ram-mohan** [comment]
@gregbenz the demo app contains changes that support this.
